src/natural_transformations.typ

```diff
@@ -0,0 +1,177 @@
+  - Is there always an $x$ such that $F iso cat(C)(x, \_)$?
+  - Is there at most (up to isomorphism) one $x$ such that $F iso cat(C)(x, \_)$?
+  - If such an $x$ exists, what properties does it have?
+
+This question might seem a bit "out of the blue": we are talking here about encoding a functor $F : cat(C) -> cat(D)$ (meaning transformations of both elements and arrows) into a _single_ element of $cat(C)$.
+
+However, given $U : Grp -> Set$ the forgetful functor, we have $Grp(ZZ, \_) iso U$.
+
+#proof[
+  Let's define $alpha : Grp(ZZ, \_) => U$. For all groups $G$, $alpha_G$ is the function:
+
+  #align(center)[$
+    Grp(ZZ, G) &  -> U G \
+       phi.alt & |-> phi.alt(1)
+  $]
+
+  First, $alpha$ is a natural transformation: indeed, given two groups $G$ and $H$, together with $phi.alt : G -> H$, the following diagram commutes:
+
+  #align(center, tikzcd[
+    ```
+    {\mathbf{Grp}(\mathbb{Z}, G)} && UG \\
+    \\
+    {\mathbf{Grp}(\mathbb{Z}, H)} && UH
+    \arrow["{\alpha_g}"', from=1-1, to=1-3]
+    \arrow["{\phi^*}"', from=1-1, to=3-1]
+    \arrow["U\phi"', from=1-3, to=3-3]
+    \arrow["{\alpha_H}"', from=3-1, to=3-3]
+    ```
+  ])
+
+  Indeed, given $f : ZZ -> G$, we have $ (U phi.alt)(alpha_G(f)) = phi.alt(f(1)) = (phi.alt compose f)(1) = alpha_G((U phi.alt)(f)) $
+
+  And, given $x in G$, because $ZZ$ is generated by a single element, there is exactly one morphism $ZZ -> G$ such that $phi.alt(1) = x$, which means that every arrow $alpha_G$ is a bijection (an isomorphism in $Set$).
+]
+
+On the contravariant case, we instead use the (contravariant) functor $cat(C)(\_, w)$: a good example is the fact $pset iso Set({0, 1}, \_)$.
+
+#proof[
+  For every $X in Set$, let's consider the following function:
+  #align(center)[$
+    alpha_X : {0, 1}^X &  -> pset(X) \
+                     f & |-> f^(-1)(1)
+  $]
+]
+
+Let's look at a few examples.
+
+In fact, $Set({0, 1}, \_) iso pset$.
+
+#proof[
+  We define $alpha : Set({0,1}, \_) => pset$ as the natural transformation $(alpha_X : {0,1}^X -> pset(X))_(X in Set)$.
+
+  Where, for each $X in Set$, $alpha_X$ is the function $f |-> f^(-1)(1)$.
+
+  First, it is clear that, for all $X$, $alpha_X$ is a bijection (therefore an isomorphism in $Set$).
+
+  And, given $f : X -> Y$, the following diagram commutes (_i.e._ for all $g in {0, 1}^X$ we have ${f(x) | g(x) = 1} = "TODO"$):
+
+  #align(center, tikzcd[
+    ```
+    {\mathbf{Set}(\{0, 1\}, X)} && {\mathcal{P}(Y)} \\
+    \\
+    {\mathbf{Set}(\{0, 1\}, Y)} && {\mathcal{P}(Y)}
+    \arrow["{\alpha_X}"', from=1-1, to=1-3]
+    \arrow["{f^*}"', from=1-1, to=3-1]
+    \arrow["{\mathcal{P}f}"', from=1-3, to=3-3]
+    \arrow["{\alpha_Y}"', from=3-1, to=3-3]
+    ```
+  ])
+]
```

Comment:
@adamAndMath wants to change `Set({0, 1}, _)` into `Set(_, {0, 1})`:
```suggestion
In fact, $Set(\_, {0, 1}) iso pset$.

#proof[
  We define $alpha : Set(\_, {0,1}) => pset$ as the natural transformation $(alpha_X : {0,1}^X -> pset(X))_(X in Set)$.

  Where, for each $X in Set$, $alpha_X$ is the function $f |-> f^(-1)(1)$.

  First, it is clear that, for all $X$, $alpha_X$ is a bijection (therefore an isomorphism in $Set$).

  And, given $f : X -> Y$, the following diagram commutes (_i.e._ for all $g in {0, 1}^X$ we have ${f(x) | g(x) = 1} = "TODO"$):

  #align(center, tikzcd[
    ```
    {\mathbf{Set}(X, \{0, 1\})} && {\mathcal{P}(Y)} \\
    \\
    {\mathbf{Set}(Y, \{0, 1\})} && {\mathcal{P}(Y)}
    \arrow["{\alpha_X}"', from=1-1, to=1-3]
    \arrow["{f^*}"', from=1-1, to=3-1]
    \arrow["{\mathcal{P}f}"', from=1-3, to=3-3]
    \arrow["{\alpha_Y}"', from=3-1, to=3-3]
    ```
  ])
]
```